sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Plugin.java

```diff
@@ -27,8 +27,6 @@
 public class Plugin extends Component {
 
 
-  private boolean supportedBySonarSource = false;
```

Comment:
Maybe I am being picky but can we remove the private constructor up a bit so there is not a large space between class definition?

Reply:
In my opinion, we can remove ample space between class definitions, but we can remove the private constructor. 
It forces us to use the factory method below it.
(the factory method checks the key, and it's a good practice to avoid  a business code in a constructor..) 

@pierre-guillot-sonarsource or @aurelien-poscia-sonarsource. What do you think about it?

Reply:
Indeed, as the code is currently, we must leave the private constructor to force users to use the factory and ultimately to have the key format validated.

Generally speaking, it is however fine for me to have simple arguments validation in the constructor.

Here the factory method was probably added, as we can't anyway validate the `key` before calling the superclass's constructor. So I would leave the code as it is.

Reply:
I agree. The private constructor should stay to only allow using the factory.